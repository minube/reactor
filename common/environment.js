import { DeviceInfo, Platform } from 'react-native';
import PKG from '../package.json';

const { NODE_ENV, REACT_APP_ENV = 'development' } = process.env;
const IS_SERVER = typeof window === 'undefined';
const IS_WEB = Platform.OS === 'web';
const IS_PRODUCTION = NODE_ENV === 'production' && NODE_ENV === REACT_APP_ENV;
const IS_MOBILE_WEB = IS_WEB && !IS_SERVER
  ? /iPhone|Android|Tablet|iPad|Mobile|iPod|Windows Phone|Blackberry/i.test(DeviceInfo.userAgent)
  : false;

if (IS_WEB && !IS_SERVER) window.reactor = { version: PKG.version };

export default {
  IS_NATIVE: !IS_WEB,
  IS_SERVER: IS_SERVER && IS_WEB,
  IS_WEB,
  IS_MOBILE_WEB,
  IS_PRODUCTION,
  IS_DEVELOPMENT: !IS_PRODUCTION,
  IS_TEST: NODE_ENV === 'test',

  NODE_ENV,

  PKG: {
    name: PKG.name,
    version: PKG.version,
  },

  REACT_APP_ENV,
};
