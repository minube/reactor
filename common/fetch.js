const DEFAULT_HEADERS = {
  'Content-Type': 'application/json',
  'X-Requested-With': 'XMLHttpRequest',
};

export default async (endpoint, props = {}) => {
  const { headers, method = 'GET', ...others } = props;

  return new Promise((resolve, reject) => {
    fetch(endpoint, {
      headers: { ...DEFAULT_HEADERS, ...headers },
      method,
      ...others,
    })
      .then(async (response) => {
        const json = await response.json();

        if (response.status >= 400) reject({ code: response.status, message: json.message }); // eslint-disable-line
        else resolve(json);
      }).catch(({ message = 'ERROR_CONNECTION', response } = {}) => {
        reject({ // eslint-disable-line
          code: response ? response.status : undefined,
          message,
        });
      });
  });
};
