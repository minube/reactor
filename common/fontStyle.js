import screenSize from './screenSize';
import THEME from './theme';

const { FONT, UNIT } = THEME;

export default (
  width,
  {
    tiny, small, large, title,
  },
) => {
  const {
    TINY, PHONE, TABLET, SMALL, REGULAR, LARGE,
  } = screenSize(width);

  let fontSize = REGULAR || LARGE ? FONT.SIZE.REGULAR : FONT.SIZE.SMALL;

  if (tiny) fontSize = REGULAR || LARGE ? FONT.SIZE.TINY : UNIT * 1.1;
  else if (small) fontSize = REGULAR || LARGE ? FONT.SIZE.SMALL : FONT.SIZE.TINY;
  else if (large) {
    fontSize = FONT.SIZE.LARGE;
    if (TINY) fontSize = FONT.SIZE.SMALL;
    if (PHONE || SMALL) fontSize = FONT.SIZE.REGULAR;
    if (TABLET || REGULAR) fontSize = UNIT * 1.8;
  } else if (title) {
    fontSize = UNIT * 4.8;
    if (TINY || PHONE || TABLET) fontSize = UNIT * 2.4;
    if (SMALL || REGULAR) fontSize = UNIT * 3;
  }

  return {
    fontSize,
    lineHeight: fontSize * (TINY || SMALL ? 1.2 : 1.3),
  };
};
