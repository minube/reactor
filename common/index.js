import STYLE from './style';
import THEME from './theme';

export {
  STYLE,
  THEME,
};
