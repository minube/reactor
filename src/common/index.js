import C from './constants';
import STYLE from './style';
import THEME from './theme';

export {
  C,
  STYLE,
  THEME,
};
