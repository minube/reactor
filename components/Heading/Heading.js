import { array, arrayOf, bool, func, number, object, oneOfType, shape, string } from 'prop-types';
import React from 'react';
import { StyleSheet, View } from 'react-native';

import { LAYOUT, SHAPE, THEME } from '../../common';
import Breadcrumbs from '../Breadcrumbs';
import Avatar from '../Avatar';
import Rating from '../Rating';
import Text from '../Text';
import styles from './Heading.style';

const { AVATAR: { SMALL }, COLOR, UNIT } = THEME;

const Heading = ({
  breadcrumbs, color, column, onBreadcrumb, rating, style, title,
  contributors: { total, label, preview } = {},
}) => (
  <View style={StyleSheet.flatten([styles.container, style])}>
    <View style={column ? styles.column : styles.row}>
      <Text
        bold
        title
        color={color}
        style={[LAYOUT.STYLE.TEXT.TITLE, styles.title]}
      >
        {title}
      </Text>
      { rating.value && <Rating {...rating} textColor={color} /> }
    </View>
    <View style={column ? styles.column : styles.row}>
      { breadcrumbs.length > 0 &&
        <Breadcrumbs color={color || COLOR.TEXT_LIGHTEN} dataSource={breadcrumbs} onPress={onBreadcrumb} /> }
      { total &&
        <View style={column ? styles.row : styles.contributors}>
          <Text bold color={color} tiny>{total}</Text>
          <Text color={color} tiny>{` ${label}`}</Text>
          <View style={StyleSheet.flatten([
              styles.avatars,
              { width: (SMALL + (UNIT / 2)) * preview.length },
            ])}
          >
            { preview.map(({ id, image }) => <Avatar key={id} small image={image} />)}
          </View>
        </View> }
    </View>
  </View>
);

Heading.propTypes = {
  breadcrumbs: arrayOf(shape({})),
  color: string,
  column: bool,
  contributors: shape({}),
  onBreadcrumb: func,
  rating: shape(SHAPE.RATING),
  style: oneOfType([array, number, object]),
  title: string.isRequired,
};

Heading.defaultProps = {
  breadcrumbs: [],
  color: undefined,
  column: false,
  contributors: undefined,
  onBreadcrumb: undefined,
  rating: {},
  style: [],
};

export default Heading;
