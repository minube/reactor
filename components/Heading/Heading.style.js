import { StyleSheet } from 'react-native';

import { STYLE, THEME } from '../../common';

const { UNIT } = THEME;

export default StyleSheet.create({
  container: {
    width: '100%',
  },

  row: {
    ...STYLE.ROW,
    minHeight: UNIT * 2,
  },

  column: {
  },

  contributors: {
    ...STYLE.ROW,
    justifyContent: 'flex-end',
    flex: 1,
  },

  avatars: STYLE.ROW,

  avatar: {
    marginLeft: UNIT / 2,
  },

  title: {
    marginRight: UNIT,
  },
});
