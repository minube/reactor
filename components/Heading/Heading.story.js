import React from 'react';
import { storiesOf } from '@storybook/react';
import { boolean, color, object, text } from '@storybook/addon-knobs/react';

import { THEME } from '../../common';
import WebView from '../WebView';
import Heading from './Heading';

const BREADCRUMBS = [
  { caption: 'Turismo', href: '/' },
  { caption: 'Gran Bretaña', href: '/' },
  { caption: 'Inglaterra', href: '/' },
  { caption: 'Londres', href: '/' },
];

const CONTRIBUTORS = {
  total: 128,
  label: 'colaborators',
  preview: [
    { id: 8, image: 'https://api.adorable.io/avatars/48/8' },
    { id: 16, image: 'https://api.adorable.io/avatars/48/16' },
    { id: 32, image: 'https://api.adorable.io/avatars/48/32' },
  ],
};

const RATING = { caption: 'valoraciones', count: 12, value: 4.5 };

storiesOf('🛠 Heading', module)
  .addWithJSX('default', () => (
    <Heading title="Actividades en España" />
  ))
  .addWithJSX('rating', () => (
    <Heading title="Actividades en España" rating={RATING} />
  ))
  .addWithJSX('breadcrumbs', () => (
    <Heading title="Actividades en España" breadcrumbs={BREADCRUMBS} />
  ))
  .addWithJSX('contributors', () => (
    <Heading contributors={CONTRIBUTORS} title="Actividades en España" />
  ))
  .addWithJSX('color', () => (
    <Heading
      breadcrumbs={BREADCRUMBS}
      color={THEME.COLOR.ACCENT}
      contributors={CONTRIBUTORS}
      title="Actividades en España"
      rating={RATING}
    />
  ))
  .addWithJSX('column', () => (
    <Heading
      breadcrumbs={BREADCRUMBS}
      contributors={CONTRIBUTORS}
      title="Actividades en España"
      rating={RATING}
      column
    />
  ))
  .addWithJSX('style', () => (
    <Heading
      color={THEME.COLOR.WHITE}
      column
      rating={RATING}
      style={{
        padding: THEME.OFFSET,
        backgroundColor: 'rgba(0,0,0,0.25)',
      }}
      title="British Museum"
    />
  ))
  .addWithJSX('🏀 Playground', () => (
    <WebView>
      <Heading
        breadcrumbs={object('breadcrumbs', BREADCRUMBS, 'heading-breadcrumbs')}
        color={color('color', THEME.COLOR.TEXT)}
        column={boolean('column', false)}
        contributors={object('contributors', CONTRIBUTORS, 'heading-contributors')}
        rating={object('rating', RATING, 'heading-rating')}
        title={text('title', 'Actividades en España')}
      />
    </WebView>
  ));

