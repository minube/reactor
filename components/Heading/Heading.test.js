import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import { THEME } from '../../common';
import Heading from './Heading';

const BREADCRUMBS = [
  { caption: 'Turismo', href: '/' },
  { caption: 'Gran Breta√±a', href: '/' },
  { caption: 'Inglaterra', href: '/' },
  { caption: 'Londres', href: '/' },
];

const CONTRIBUTORS = {
  total: 128,
  label: 'colaborators',
  preview: [
    { id: 8, image: 'https://api.adorable.io/avatars/48/8' },
    { id: 16, image: 'https://api.adorable.io/avatars/48/16' },
    { id: 32, image: 'https://api.adorable.io/avatars/48/32' },
  ],
};

const RATING = { caption: 'valoraciones', count: 12, value: 4.5 };

describe('<Heading>', () => {
  it('renders', () => {
    const tree = renderer.create(<Heading />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {title}', () => {
    const tree = renderer.create(<Heading title="Hello" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {rating}', () => {
    const tree = renderer.create(<Heading title="Hello" rating={RATING} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {breadcrumbs}', () => {
    const tree = renderer.create(<Heading title="Hello" breadcrumbs={BREADCRUMBS} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {contributors}', () => {
    const tree = renderer.create(<Heading title="Hello" contributors={CONTRIBUTORS} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {color}', () => {
    const tree = renderer.create(<Heading title="Hello" color={THEME.COLOR.ACCENT} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {column}', () => {
    const tree = renderer.create(<Heading title="Hello" column rating={RATING} contributors={CONTRIBUTORS} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {style}', () => {
    const style = {
      padding: THEME.OFFSET,
      backgroundColor: 'rgba(0,0,0,0.25)',
    };
    const tree = renderer.create(<Heading title="Hello" style={style} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders in DOM environment', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Heading />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
});

