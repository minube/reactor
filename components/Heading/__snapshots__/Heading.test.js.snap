// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading> renders 1`] = `
<View
  style={
    Object {
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "minHeight": undefined,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      numberOfLines={undefined}
      style={
        Object {
          "color": "#4a4a4a",
          "fontFamily": undefined,
          "fontSize": 30,
          "fontWeight": "bold",
          "lineHeight": 39,
          "marginRight": 10,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "minHeight": undefined,
      }
    }
  />
</View>
`;

exports[`<Heading> when {breadcrumbs} 1`] = `
<View
  style={
    Object {
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "minHeight": undefined,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      numberOfLines={undefined}
      style={
        Object {
          "color": "#4a4a4a",
          "fontFamily": undefined,
          "fontSize": 30,
          "fontWeight": "bold",
          "lineHeight": 39,
          "marginRight": 10,
        }
      }
    >
      Hello
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "minHeight": undefined,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
        }
      }
    >
      <Text
        accessibilityRole="link"
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        href="/"
        numberOfLines={undefined}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onPress={[Function]}
        style={
          Object {
            "color": "#9b9b9b",
            "fontFamily": undefined,
            "fontSize": 14,
            "lineHeight": 18.2,
            "marginRight": 4.444444444444445,
          }
        }
      >
        Turismo ·
      </Text>
      <Text
        accessibilityRole="link"
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        href="/"
        numberOfLines={undefined}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onPress={[Function]}
        style={
          Object {
            "color": "#9b9b9b",
            "fontFamily": undefined,
            "fontSize": 14,
            "lineHeight": 18.2,
            "marginRight": 4.444444444444445,
          }
        }
      >
        Gran Bretaña ·
      </Text>
      <Text
        accessibilityRole="link"
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        href="/"
        numberOfLines={undefined}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onPress={[Function]}
        style={
          Object {
            "color": "#9b9b9b",
            "fontFamily": undefined,
            "fontSize": 14,
            "lineHeight": 18.2,
            "marginRight": 4.444444444444445,
          }
        }
      >
        Inglaterra ·
      </Text>
      <Text
        accessibilityRole="link"
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        href="/"
        numberOfLines={undefined}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onPress={[Function]}
        style={
          Object {
            "color": "#9b9b9b",
            "fontFamily": undefined,
            "fontSize": 14,
            "lineHeight": 18.2,
            "marginRight": 4.444444444444445,
          }
        }
      >
        Londres
      </Text>
    </View>
  </View>
</View>
`;

exports[`<Heading> when {color} 1`] = `
<View
  style={
    Object {
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "minHeight": undefined,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      numberOfLines={undefined}
      style={
        Object {
          "color": "#f69a37",
          "fontFamily": undefined,
          "fontSize": 30,
          "fontWeight": "bold",
          "lineHeight": 39,
          "marginRight": 10,
        }
      }
    >
      Hello
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "minHeight": undefined,
      }
    }
  />
</View>
`;

exports[`<Heading> when {column} 1`] = `
<View
  style={
    Object {
      "width": "100%",
    }
  }
>
  <View
    style={Object {}}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      numberOfLines={undefined}
      style={
        Object {
          "color": "#4a4a4a",
          "fontFamily": undefined,
          "fontSize": 30,
          "fontWeight": "bold",
          "lineHeight": 39,
          "marginRight": 10,
        }
      }
    >
      Hello
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={undefined}
        style={
          Object {
            "color": "#10c0bc",
            "fontFamily": undefined,
            "fontSize": 16,
            "lineHeight": 16,
            "marginRight": 2.5,
          }
        }
      >
        ★
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={undefined}
        style={
          Object {
            "color": "#10c0bc",
            "fontFamily": undefined,
            "fontSize": 16,
            "lineHeight": 16,
            "marginRight": 2.5,
          }
        }
      >
        ★
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={undefined}
        style={
          Object {
            "color": "#10c0bc",
            "fontFamily": undefined,
            "fontSize": 16,
            "lineHeight": 16,
            "marginRight": 2.5,
          }
        }
      >
        ★
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={undefined}
        style={
          Object {
            "color": "#10c0bc",
            "fontFamily": undefined,
            "fontSize": 16,
            "lineHeight": 16,
            "marginRight": 2.5,
          }
        }
      >
        ★
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={undefined}
        style={
          Object {
            "color": "#10c0bc",
            "fontFamily": undefined,
            "fontSize": 16,
            "lineHeight": 16,
            "marginRight": 2.5,
          }
        }
      >
        ☆
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={undefined}
        style={
          Object {
            "color": "#9b9b9b",
            "fontFamily": undefined,
            "fontSize": 12,
            "fontWeight": "bold",
            "lineHeight": 15.600000000000001,
          }
        }
      >
         4.5/5
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={undefined}
        style={
          Object {
            "color": "#9b9b9b",
            "fontFamily": undefined,
            "fontSize": 12,
            "lineHeight": 15.600000000000001,
          }
        }
      >
         (12 valoraciones)
      </Text>
    </View>
  </View>
  <View
    style={Object {}}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "minHeight": undefined,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={undefined}
        style={
          Object {
            "color": "#4a4a4a",
            "fontFamily": undefined,
            "fontSize": 12,
            "fontWeight": "bold",
            "lineHeight": 15.600000000000001,
          }
        }
      >
        128
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={undefined}
        style={
          Object {
            "color": "#4a4a4a",
            "fontFamily": undefined,
            "fontSize": 12,
            "lineHeight": 15.600000000000001,
          }
        }
      >
         colaborators
      </Text>
      <View
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingLeft": 5,
            "width": 75,
          }
        }
      >
        <View
          style={
            Object {
              "alignContent": "center",
              "alignItems": "center",
              "backgroundColor": "#f4f4f4",
              "borderRadius": 10,
              "height": 20,
              "justifyContent": "center",
              "width": 20,
            }
          }
        >
          <Image
            onLoad={[Function]}
            resizeMode="cover"
            source={
              Object {
                "uri": "https://api.adorable.io/avatars/48/8",
              }
            }
            style={
              Object {
                "borderRadius": 10,
                "height": 20,
                "width": 20,
              }
            }
          />
          <ActivityIndicator
            animating={true}
            color="white"
            hidesWhenStopped={true}
            size="large"
            style={
              Object {
                "position": "absolute",
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "alignContent": "center",
              "alignItems": "center",
              "backgroundColor": "#f4f4f4",
              "borderRadius": 10,
              "height": 20,
              "justifyContent": "center",
              "width": 20,
            }
          }
        >
          <Image
            onLoad={[Function]}
            resizeMode="cover"
            source={
              Object {
                "uri": "https://api.adorable.io/avatars/48/16",
              }
            }
            style={
              Object {
                "borderRadius": 10,
                "height": 20,
                "width": 20,
              }
            }
          />
          <ActivityIndicator
            animating={true}
            color="white"
            hidesWhenStopped={true}
            size="large"
            style={
              Object {
                "position": "absolute",
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "alignContent": "center",
              "alignItems": "center",
              "backgroundColor": "#f4f4f4",
              "borderRadius": 10,
              "height": 20,
              "justifyContent": "center",
              "width": 20,
            }
          }
        >
          <Image
            onLoad={[Function]}
            resizeMode="cover"
            source={
              Object {
                "uri": "https://api.adorable.io/avatars/48/32",
              }
            }
            style={
              Object {
                "borderRadius": 10,
                "height": 20,
                "width": 20,
              }
            }
          />
          <ActivityIndicator
            animating={true}
            color="white"
            hidesWhenStopped={true}
            size="large"
            style={
              Object {
                "position": "absolute",
              }
            }
          />
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`<Heading> when {contributors} 1`] = `
<View
  style={
    Object {
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "minHeight": undefined,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      numberOfLines={undefined}
      style={
        Object {
          "color": "#4a4a4a",
          "fontFamily": undefined,
          "fontSize": 30,
          "fontWeight": "bold",
          "lineHeight": 39,
          "marginRight": 10,
        }
      }
    >
      Hello
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "minHeight": undefined,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "flex-end",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={undefined}
        style={
          Object {
            "color": "#4a4a4a",
            "fontFamily": undefined,
            "fontSize": 12,
            "fontWeight": "bold",
            "lineHeight": 15.600000000000001,
          }
        }
      >
        128
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={undefined}
        style={
          Object {
            "color": "#4a4a4a",
            "fontFamily": undefined,
            "fontSize": 12,
            "lineHeight": 15.600000000000001,
          }
        }
      >
         colaborators
      </Text>
      <View
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingLeft": 5,
            "width": 75,
          }
        }
      >
        <View
          style={
            Object {
              "alignContent": "center",
              "alignItems": "center",
              "backgroundColor": "#f4f4f4",
              "borderRadius": 10,
              "height": 20,
              "justifyContent": "center",
              "width": 20,
            }
          }
        >
          <Image
            onLoad={[Function]}
            resizeMode="cover"
            source={
              Object {
                "uri": "https://api.adorable.io/avatars/48/8",
              }
            }
            style={
              Object {
                "borderRadius": 10,
                "height": 20,
                "width": 20,
              }
            }
          />
          <ActivityIndicator
            animating={true}
            color="white"
            hidesWhenStopped={true}
            size="large"
            style={
              Object {
                "position": "absolute",
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "alignContent": "center",
              "alignItems": "center",
              "backgroundColor": "#f4f4f4",
              "borderRadius": 10,
              "height": 20,
              "justifyContent": "center",
              "width": 20,
            }
          }
        >
          <Image
            onLoad={[Function]}
            resizeMode="cover"
            source={
              Object {
                "uri": "https://api.adorable.io/avatars/48/16",
              }
            }
            style={
              Object {
                "borderRadius": 10,
                "height": 20,
                "width": 20,
              }
            }
          />
          <ActivityIndicator
            animating={true}
            color="white"
            hidesWhenStopped={true}
            size="large"
            style={
              Object {
                "position": "absolute",
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "alignContent": "center",
              "alignItems": "center",
              "backgroundColor": "#f4f4f4",
              "borderRadius": 10,
              "height": 20,
              "justifyContent": "center",
              "width": 20,
            }
          }
        >
          <Image
            onLoad={[Function]}
            resizeMode="cover"
            source={
              Object {
                "uri": "https://api.adorable.io/avatars/48/32",
              }
            }
            style={
              Object {
                "borderRadius": 10,
                "height": 20,
                "width": 20,
              }
            }
          />
          <ActivityIndicator
            animating={true}
            color="white"
            hidesWhenStopped={true}
            size="large"
            style={
              Object {
                "position": "absolute",
              }
            }
          />
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`<Heading> when {rating} 1`] = `
<View
  style={
    Object {
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "minHeight": undefined,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      numberOfLines={undefined}
      style={
        Object {
          "color": "#4a4a4a",
          "fontFamily": undefined,
          "fontSize": 30,
          "fontWeight": "bold",
          "lineHeight": 39,
          "marginRight": 10,
        }
      }
    >
      Hello
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={undefined}
        style={
          Object {
            "color": "#10c0bc",
            "fontFamily": undefined,
            "fontSize": 16,
            "lineHeight": 16,
            "marginRight": 2.5,
          }
        }
      >
        ★
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={undefined}
        style={
          Object {
            "color": "#10c0bc",
            "fontFamily": undefined,
            "fontSize": 16,
            "lineHeight": 16,
            "marginRight": 2.5,
          }
        }
      >
        ★
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={undefined}
        style={
          Object {
            "color": "#10c0bc",
            "fontFamily": undefined,
            "fontSize": 16,
            "lineHeight": 16,
            "marginRight": 2.5,
          }
        }
      >
        ★
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={undefined}
        style={
          Object {
            "color": "#10c0bc",
            "fontFamily": undefined,
            "fontSize": 16,
            "lineHeight": 16,
            "marginRight": 2.5,
          }
        }
      >
        ★
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={undefined}
        style={
          Object {
            "color": "#10c0bc",
            "fontFamily": undefined,
            "fontSize": 16,
            "lineHeight": 16,
            "marginRight": 2.5,
          }
        }
      >
        ☆
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={undefined}
        style={
          Object {
            "color": "#9b9b9b",
            "fontFamily": undefined,
            "fontSize": 12,
            "fontWeight": "bold",
            "lineHeight": 15.600000000000001,
          }
        }
      >
         4.5/5
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={undefined}
        style={
          Object {
            "color": "#9b9b9b",
            "fontFamily": undefined,
            "fontSize": 12,
            "lineHeight": 15.600000000000001,
          }
        }
      >
         (12 valoraciones)
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "minHeight": undefined,
      }
    }
  />
</View>
`;

exports[`<Heading> when {style} 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgba(0,0,0,0.25)",
      "padding": 16,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "minHeight": undefined,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      numberOfLines={undefined}
      style={
        Object {
          "color": "#4a4a4a",
          "fontFamily": undefined,
          "fontSize": 30,
          "fontWeight": "bold",
          "lineHeight": 39,
          "marginRight": 10,
        }
      }
    >
      Hello
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "minHeight": undefined,
      }
    }
  />
</View>
`;

exports[`<Heading> when {title} 1`] = `
<View
  style={
    Object {
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "minHeight": undefined,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      numberOfLines={undefined}
      style={
        Object {
          "color": "#4a4a4a",
          "fontFamily": undefined,
          "fontSize": 30,
          "fontWeight": "bold",
          "lineHeight": 39,
          "marginRight": 10,
        }
      }
    >
      Hello
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "minHeight": undefined,
      }
    }
  />
</View>
`;
